ExampleApp/ExampleApp/Screens/Characteristics/ViewController/CharacterisiticsViewController.swift

```diff
@@ -0,0 +1,175 @@
+import RxBluetoothKit
+import RxSwift
+import RxCocoa
+import UIKit
+
+final class CharacteristicsViewController: UIViewController, CustomView {
+
+    typealias ViewClass = BaseView
+
+    typealias CharacteristicsDataSource = TableViewDataSource<[Characteristic], CharacteristicsViewModelItem>
+
+    private let viewModel: CharacteristicsViewModelType
+
+    private let dataSource: CharacteristicsDataSource
+
+    private let disposeBag: DisposeBag = DisposeBag()
+
+    init(with dataSource: CharacteristicsDataSource, viewModel: CharacteristicsViewModelType) {
+        self.dataSource = dataSource
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+        view.backgroundColor = .white
+    }
+
+    @available(*, unavailable)
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func loadView() {
+        self.view = ViewClass()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupTableView()
+        setDataSourceRefreshBlock()
+        bindViewModel()
+    }
+
+    private func setupTableView() {
+        customView.setTableView(dataSource: dataSource, delegate: self)
+        registerCells()
+    }
+
+    private func setDataSourceRefreshBlock() {
+        self.dataSource.refreshDataBlock = { [weak self] in
+            self?.customView.refreshTableView()
+        }
+    }
+
+    private func bindViewModel() {
+        subscribeViewModelOutputs()
+        dataSource.bindItemsObserver(to: viewModel.characteristicsOutput)
+    }
+
+    private func subscribeViewModelOutputs() {
```

Comment:
isn't calling functions redundant in this case? `subscribeDataUpdateOutput` and `subscribeAlertTriggerOutput` are short and naming these functions don't increase understanding of the code